internal/services/storage/storage_account_resource.go

```diff
@@ -1482,21 +1472,70 @@ func resourceStorageAccountCreate(d *pluginsdk.ResourceData, meta interface{}) e
 	if account.Model == nil {
 		return fmt.Errorf("retrieving %s: `model` was nil", id)
 	}
-	if err := storageClient.AddToCache(id, *account.Model); err != nil {
+	if err := storageUtils.AddToCache(id, *account.Model); err != nil {
 		return fmt.Errorf("populating cache for %s: %+v", id, err)
 	}
 
-	dataPlaneAccount, err := storageClient.FindAccount(ctx, id.SubscriptionId, id.StorageAccountName)
-	if err != nil {
-		return fmt.Errorf("retrieving %s: %+v", id, err)
-	}
-	if dataPlaneAccount == nil {
-		return fmt.Errorf("unable to locate %q", id)
-	}
-
 	supportLevel := availableFunctionalityForAccount(accountKind, accountTier, replicationType)
-	if err := waitForDataPlaneToBecomeAvailableForAccount(ctx, storageClient, dataPlaneAccount, supportLevel); err != nil {
-		return fmt.Errorf("waiting for the Data Plane for %s to become available: %+v", id, err)
+	// Start of Data Plane access - this entire block can be removed for 5.0, as the data_plane_available flag becomes redundant at that time.
```

Comment:
Just to clarify, will we be removing the `data_plane_available` feature flag in 5.0?

Reply:
Correct. There's no mechanism for deprecating Provider features at the moment, and such functionality is beyond the scope of this PR. 5.0 will have no overlap of Resource Manager and Data Plane API usage in resources so this flag will be redundant at that time.